tests/helpers/test_frame.py

```diff
@@ -36,6 +36,39 @@ async def test_extract_frame_integration(caplog):
     assert found_frame == correct_frame
 
 
+async def test_extract_frame_integration_with_excluded_intergration(caplog):
+    """Test extracting the current frame from integration context."""
+    correct_frame = Mock(
+        filename="/home/dev/homeassistant/components/mdns/light.py",
+        lineno="23",
+        line="self.light.is_on",
+    )
+    with patch(
+        "homeassistant.helpers.frame.extract_stack",
+        return_value=[
+            Mock(
+                filename="/home/dev/homeassistant/core.py",
+                lineno="23",
+                line="do_something()",
+            ),
+            correct_frame,
+            Mock(
+                filename="/home/dev/homeassistant/components/zeroconf/usage.py",
+                lineno="23",
+                line="self.light.is_on",
+            ),
+            Mock(filename="/home/dev/mdns/lights.py", lineno="2", line="something()",),
+        ],
+    ):
+        found_frame, integration, path = frame.get_integration_frame(
+            exclude_integrations={"zeroconf"}
+        )
+
+    assert integration == "mdns"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)